### DISCUSSION TITLE
loop over array and use elements in useReadContract hook

### DISCUSSION THREAD

**Discussion by vantwoutmaarten** [wevm/wagmi#3728] [General]
I was wondering about a common scenario, where you want to loop over an array and use the elements as an argument for useReadContract. 

Since it is a hook where in a React component would you put this or in what way should this problem be tackled.

example:

const res = [];
for(e : elements){
const data = useReadContract(....args: [e]....)
res.push(data)
}

**Comment by tmm** [comment]
https://wagmi.sh/react/api/hooks/useReadContracts#usereadcontracts

**Reply by vantwoutmaarten** [reply]
Thanks for the quick reply! but its not directly showing when you want to do multiple different calls in a loop. I guess maybe the loop can be done in the args of useReadContracts()
